src/Rings/MPolyMap/flattenings.jl

```diff
@@ -468,12 +468,9 @@ function flatten(R::MPolyQuoRing; cached::Bool=false)
 end
 
 function (phi::RingFlattening)(I::MPolyIdeal)
-  if haskey(flat_counterparts(phi), I)
-    return flat_counterparts(phi)[I][1]
-  end
-  I_flat = ideal(codomain(phi), elem_type(codomain(phi))[phi(g) for g in gens(I)])
-  flat_counterparts(phi)[I] = (I_flat, phi)
```

Comment:
Note that I changed this, instead of storing a pair, only store `I_flat`. The second part was not used anywhere.

Reply:
Yes, this seems weird indeed. The new version is more clear.